### PR TITLE
Android freeze fix

### PR DESCRIPTION
**Pull request by nirvn**
Trying to get us below the acceptable ANR rate on play store. Bonus: people will see a progress bar instead of a frozen app :wink:

Tested, we can nice indeterminate progress bar when importing 50mb geotiffs instead of a frozen app.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 680c45a8faba455ac67f6ce596115572dc737f1b: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4939-66093c7e7d083264d43a398656e32a9dbc2cdda2-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4939-66093c7e7d083264d43a398656e32a9dbc2cdda2-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4939-66093c7e7d083264d43a398656e32a9dbc2cdda2-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4939-66093c7e7d083264d43a398656e32a9dbc2cdda2-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4939-66093c7e7d083264d43a398656e32a9dbc2cdda2-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4939-66093c7e7d083264d43a398656e32a9dbc2cdda2-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4939-66093c7e7d083264d43a398656e32a9dbc2cdda2-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4939-66093c7e7d083264d43a398656e32a9dbc2cdda2-x86-android.apk)
</details>

File: platform/android/res/values/strings.xml [hunk 1]
```diff
@@ -64,4 +64,5 @@
     <string name="operation_unsupported">Operation Unsupported</string>
     <string name="import_operation_unsupported">Your device does not support this import operation.</string>
     <string name="export_operation_unsupported">Your device does not support this export operation.</string>
+    <string name="processing_message">Processing...</string>
```
Comment (m-kuhn) [inline platform/android/res/values/strings.xml:67]:
```suggestion
    <string name="processing_message">Processing…</string>
```

:scream:

File: src/core/platforms/platformutilities.h [hunk 2]
```diff
@@ -96,10 +96,12 @@ class QFIELD_CORE_EXPORT PlatformUtilities : public QObject
     virtual QString systemLocalDataLocation( const QString &subDir ) const;
 
     /**
-     * \returns a QGIS project which should be opened. This depends on platform specific
-     *          implementations. This is currently only available on Android via Intents.
+     * \returns TRUE if a QGIS project which should be opened at launch.
```
Comment (m-kuhn) [inline src/core/platforms/platformutilities.h:99]: This sounds grammatically incorrect

**Comment by nirvn** [comment]
@m-kuhn , comments addressed, thanks.

**Review by m-kuhn — APPROVED** [review]

**Comment by qfield-fairy** [comment]
The backport to `release-3_1` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 128eae381... Add missing cups dev package to Android docker setup
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging src/core/qgismobileapp.cpp
[backport-4939-to-release-3_1 4e5a481c4] Fix Android freezes when processing intent and sending folders
 Author: Mathieu Pellerin <nirvn.asia@gmail.com>
 Date: Sat Jan 13 11:08:50 2024 +0700
 8 files changed, 243 insertions(+), 152 deletions(-)
Auto-merging .docker/android_dev/Dockerfile
CONFLICT (content): Merge conflict in .docker/android_dev/Dockerfile

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-release-3_1 release-3_1
# Navigate to the new working tree
cd .worktrees/backport-release-3_1
# Create a new branch
git switch --create backport-4939-to-release-3_1
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 0beef2143c96f6099e1828e2c9b60bf761388c7a,128eae381ef67cb5319ae01b94d39dd36d4af5b6
# Push it to GitHub
git push --set-upstream origin backport-4939-to-release-3_1
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-release-3_1
```
Then, create a pull request where the `base` branch is `release-3_1` and the `compare`/`head` branch is `backport-4939-to-release-3_1`.

**Pull request merged**